docs/DeveloperGuide.md

```diff
@@ -1,14 +1,157 @@
+and `vacancy: `2.
+
+Step 2: The system calls `App Controller` and prompts "Enter Command: ".
+
+Step 3: The user enters `assign employee`. This command will call `assignEmployee()` in `App Controller`, which then 
+causes the system to output "Enter Employee name to assign: ".
+
+Step 4: The user inputs "Eve", which calls the`getName()` method in `Employee` to check whether `Eve` exists in the database.
+If `Eve` does not exist in the database, the system will output "Employee not found", and the method ends here. Since `Eve`
+exist in the database, the system proceeds to output "Enter Shift date (in dd/MM/yyyy)".
+
+Step 5: The user inputs "04/04/2021" and this is stored as a temporary `shiftDate` object in the method, where the `LocalDate.parse()` method will parse the `String` input using the `DateTimeFormatter` into a `LocalDate` object.
+The system will then output "Enter Shift index:".
+
+Step 6: The user inputs "3" which is temporarily stored as `shiftIndex`, and the system proceeds to verify that the `Shift` object with this particular `shiftDate` and `shiftIndex` exists.
+The system outputs "Shift not found" if the particular `Shift` object does not exist. Since this `Shift` exists, the system
+calls the `assignEmployee()` method in `Shift` to assign `Eve` to this shift by adding `Eve` into the ArrayList of `employees` stored in the `Shift` object. This causes the `vacancy` in this current `Shift` object to decrement by 1.
+If the `vacancy` is 0, the system outputs "Shift is full!". Since there is still 2 vacancies for this shift, the system will instead
+output "Employee Eve assigned." and `vacancy` becomes 1.
+
+Given below is a sequence diagram of the Assign Employee feature.
+![assignEmployee.PNG](assignEmployee.png) 
+
+### Saving of Data
+#### Implementation
+The ArrayList of Employee objects is first formatted by adding the '#' symbol as a delimiter between the objects and their attributes.
+<br>For example, `cactus#23/03/2021`, which saves the employee name `cactus` and the employee schedule `23/03/2021`.
+<br>This allows data of objects to be easily written into text files and then loaded.
+<br>The ArrayList of Shift objects is saved using the same method.
+<br>For example, `28/03/2021#5#1#cactus#mario`, which saves a shift on `28/03/2021`, with a vacancy of `5`, and shift index of `1`.
+The employees on that shift are`cactus` and `mario`.
+
+Employee and shift details are saved on separate text files "employees.text" and "shifts.text".
+A FileWriter is used to write formatted data from the Employee and Shift classes, with "#" as the delimiter.
+<br>The text files are overwritten with the most recent data whenever the user exits the program by inputting the command `quit`.
+
+### Loading of Data
+#### Implementation
+The FileManager Object is first initialized at the start of the program, and loads both
+the "employees.txt" and "shifts.txt" files.
```

Comment:
What happens if the files are not found? Will it throw an error, create a new file, or..?